model/trigger/task.go

```diff
@@ -363,3 +368,131 @@ func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.No
 
 	return n, errors.Wrap(rec.Insert(), "failed to process regression trigger")
 }
+
+func isFailedTaskStatus(status string) bool {
+	return status == evergreen.TaskFailed || status == evergreen.TaskSystemFailed || status == evergreen.TaskTestTimedOut
+}
+
+func isTestRegression(oldStatus, newStatus string) bool {
+	switch oldStatus {
+	case evergreen.TestSkippedStatus, evergreen.TestSucceededStatus,
+		evergreen.TestSilentlyFailedStatus:
+		if newStatus == evergreen.TestFailedStatus {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isTaskRegression(oldStatus, newStatus string) bool {
```

Comment:
To make this more closely aligned with the task status model, I think it should check the tasks themselves. Success/fail will be in the status, and different types of failure will be in the details

Reply:
See my reply to brian at the bottom of this regarding task statuses.